Logger/LogFactory.cs

```diff
@@ -2,10 +2,23 @@
 {
     public class LogFactory
     {
-        public BaseLogger CreateLogger(string className)
+        private string? fileName;
+        public string FileName { get => fileName; set => fileName = value; }
+        public BaseLogger? CreateLogger(string className)
         {
-            
-            return null;
+            if(string.IsNullOrEmpty(FileName))
+            {
+                return null;
+            }
+            else
+            {
+                FileLogger fl = new FileLogger(FileName, className);
```

Comment:
```suggestion
                FileLogger fl = new FileLogger(_filePath) { ClassName = className };
```
LogFactory should not be using a property for the file path (it should be a private field. An object initializer should also be used to set ClassName.